projects/wizard_of_wikipedia/wizard_transformer_ranker/wizard_transformer_ranker.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.agents.transformer.transformer import TransformerRankerAgent
+from .wizard_dict import WizardDictAgent
+
+import numpy as np
+import torch
+
+
+SOC_TOKEN = '__SOC__'
+
+
+class WizardTransformerRankerAgent(TransformerRankerAgent):
+    @staticmethod
+    def dictionary_class():
+        return WizardDictAgent
+
+    @classmethod
+    def add_cmdline_args(cls, argparser):
+        """Add command-line arguments specifically for this agent."""
+        super(WizardTransformerRankerAgent, cls).add_cmdline_args(argparser)
+        agent = argparser.add_argument_group('Wizard Transformer Ranker Arguments')
+        agent.add_argument(
+            '--use-knowledge', type='bool', default=True,
+            help='use knowledge field instead of personas'
+        )
+        agent.add_argument(
+            '--knowledge-dropout', type=float, default=0.7,
+            help='dropout some knowledge during training'
+        )
+        agent.add_argument(
+            '--chosen-sentence', type='bool', default=False,
+            help='instead of using all knowledge, use gold'
+                 'label, i.e. the chosen sentence'
+        )
+        agent.add_argument(
+            '--knowledge-truncate', type=int, default=50,
+            help='truncate knowledge to this length'
+        )
+        agent.add_argument(
+            '--legacy', type='bool', default=False,
+            help='legacy model'
+        )
+        argparser.set_defaults(
+            learningrate=0.0008,
+            eval_candidates='inline',
+            candidates='batch',
+            lr_factor=1,
+            add_p1_after_newln=False,
+            delimiter=' '
```

Comment:
above we set `delimiter` to `__SOC__` - why is this one different?

Reply:
`__SOC__` was only used in legacy models, because we were finetuning on top of a model we trained this way. Since this isn't standard in ParlAI I think we should only do this in the case of legacy models.